File: src/crud_feature/application/use_cases.py
from __future__ import annotations

from ..domain.entities import CrudFeature
from ..domain.repositories import CrudFeatureRepository


class CreateCrudFeatureUseCase:
    def __init__(self, repository: CrudFeatureRepository) -> None:
        self._repository = repository

    def execute(self, entity_id: str, name: str) -> CrudFeature:
        entity = CrudFeature(id=entity_id, name=name)
        return self._repository.save(entity)

File: src/crud/application/use-cases.ts
import type { Crud } from '../domain/entities';
import type { CrudRepository } from '../domain/repositories';

export class CreateCrudUseCase {
  constructor(private readonly repository: CrudRepository) {}

  async execute(id: string, name: string): Promise<Crud> {
    const entity: Crud = { id, name };
    return this.repository.save(entity);
  }
}
